pkg/cmd/render/render.go

```diff
@@ -275,6 +296,55 @@ func mustReadTemplateFile(fname string) options.Template {
 	return options.Template{FileName: fname, Content: bs}
 }
 
+func discoverCIDRs(clusterConfigFileData []byte, renderConfig *TemplateData) error {
+	if err := discoverCIDRsFromNetwork(clusterConfigFileData, renderConfig); err != nil {
+		if err = discoverCIDRsFromClusterAPI(clusterConfigFileData, renderConfig); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func discoverCIDRsFromNetwork(clusterConfigFileData []byte, renderConfig *TemplateData) error {
+	configJson, err := yaml.YAMLToJSON(clusterConfigFileData)
```

Comment:
the next 11 lines are the same for both discoverCIDRsFromNetwork and discoverCIDRsFromClusterAPI, why not a separate function that returns and *Unstructured and error?